src/services/codefixes/fixUnusedIdentifier.ts

```diff
@@ -121,9 +121,20 @@ namespace ts.codefix {
                 break;
 
             case SyntaxKind.Parameter:
-                const functionDeclaration = <FunctionDeclaration>parent.parent;
-                if (functionDeclaration.parameters.length === 1) {
-                    changes.deleteNode(sourceFile, parent);
+                const oldFunction = parent.parent;
+                if (isArrowFunction(oldFunction) && oldFunction.parameters.length === 1) {
+                    const newFunction = updateArrowFunction(
+                        oldFunction,
+                        oldFunction.modifiers,
+                        oldFunction.typeParameters,
+                        /*parameters*/ undefined,
+                        oldFunction.type,
+                        oldFunction.equalsGreaterThanToken,
+                        oldFunction.body);
+
+                    suppressLeadingAndTrailingTrivia(newFunction);

```

Comment:
Could you add a comment why this is done?

Reply:
Done.

Reply:
I see the comment above `newFunction`, but I was talking about the line `suppressLeadingAndTrailingTrivia`.

Reply:
Added